### PR TITLE
PLAT-6799 Update Dockerfile and .dockerignore

### PR DESCRIPTION
**Pull request by mirr254**
## Ticket

[PLAT-6799](https://crowdbotics.atlassian.net/browse/PLAT-6799)
_Related ticket:_ [PLAT-5484](https://crowdbotics.atlassian.net/browse/PLAT-5484)
_Related PR:_ [Update Django Scaffold Docker image](https://github.com/crowdbotics/django-scaffold/pull/342)

## Type of PR

- [ ] Bugfix
- [ ] New feature
- [x] Minor changes

## Changes introduced

Update `Dockerfile` to use a base image created in the `project-deploy/docker/cb-django/Dockerfile` to improve overall build speeds.

## Test and review

_Describe how a reviewer should test your PR_

* Pull the base image `crowdbotics/cb-django:3.8-slim-buster` from dockerhub .
* Checkout the PLAT-6799 branch
* Use `cookiecutter` to create a project from the template and cd into the project source folder
* Run `docker build -t <image-name:tag> --build-arg=SECRET_ENV=<some_super_secret_key> -f Dockerfile  .` 

Once the build succeeds the image should be available when you run `docker images` or `docker image ls`.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/.dockerignore [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
+docs/_build/
+
+# PyBuilder
+target/
+
+# Docker
+Dockerfile
+docker-compose.yml
+docker-compose.override.yml
+docker/docker-compose.prod.yml
```
Comment (driverdan) [inline {{cookiecutter.project_slug}}/.dockerignore:72]: Why is this excluding docker related files?

**Review by driverdan — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_slug}}/.dockerignore (hunk 1 above)
Comment (mirr254) [reply {{cookiecutter.project_slug}}/.dockerignore:72]: @driverdan The image built doesn't necessarily need to have the docker-related files since we won't actually be running `docker *` commands from the containers created from it.

Comment (driverdan) [reply {{cookiecutter.project_slug}}/.dockerignore:72]: Oh of course, I misread this and thought it was .gitignore, not .dockerignore 🤦

**Review by driverdan — APPROVED** [review]

**Comment by driverdan** [comment]
I tested this out and it's working. For some reason it doesn't seem to speed up deploys which is disappointing but it still something we should use.
